src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -388,51 +424,55 @@ public static boolean isPathIdentifier(String name) {
     return name.contains(MODULE_SLASH);
   }
 
-  private static ImmutableList<String> createRootPaths(
+  private static ImmutableList<InputPathData> createRootPaths(
       Iterable<String> roots, PathResolver resolver) {
-    ImmutableList.Builder<String> builder = ImmutableList.builder();
+    ImmutableList.Builder<InputPathData> builder = ImmutableList.builder();
     for (String root : roots) {
-      builder.add(ModuleNames.escapePath(resolver.apply(root)));
+      builder.add(new InputPathData(ModuleNames.escapePath(resolver.apply(root))));
     }
     return builder.build();
   }
 
-  private static ImmutableSet<String> resolvePaths(
-      Iterable<String> names, ImmutableList<String> roots) {
-    HashSet<String> resolved = new HashSet<>();
+  private static ImmutableMap<String, InputPathData> resolvePaths(
```

Comment:
Could you add a Javadoc comment explaining what the keys and values of the return value represent?